cirq/work/observable_measurement_data.py

```diff
@@ -423,13 +445,28 @@ def variance(self, setting: InitObsSetting, *, atol: float = 1e-8):
             raise ValueError("No measurements")
         self._validate_setting(setting, what='variance')
 
-        _, var = _stats_from_measurements(
+        mean, var = _stats_from_measurements(
             bitstrings=self.bitstrings,
             qubit_to_index=self._qubit_to_index,
             observable=setting.observable,
             atol=atol,
         )
 
+        if self._readout_calibration is not None:
+            a = mean
+            if np.isclose(a, 0, atol=atol):
+                return np.inf
+            var_a = var
+            ro_setting = _setting_to_z_observable(setting)
+            b = self._readout_calibration.mean(ro_setting)
+            if np.isclose(b, 0, atol=atol):
+                return np.inf
+            var_b = self._readout_calibration.variance(ro_setting)
+            f = a / b
+
+            # assume cov(a,b) = 0, otherwise there would be another term.
+            var = f ** 2 * (var_a / (a ** 2) + var_b / (b ** 2))
```

Comment:
can you add a reference or derivation for this formula?

Reply:
added